daffodil-test/src/test/scala/org/apache/daffodil/section07/external_variables/TestExternalVariables.scala

```diff
@@ -56,6 +56,10 @@ class TestExternalVariables {
     runner.runOneTest("override_define_vars_05")
   }
 
+  @Test def test_override_define_vars_06(): Unit = {
+    runner.runOneTest("override_define_vars_06")
```

Comment:
Can we also add this same test  but with the config no set so we test the logic that if a variable is external, but no external variable is set then it uses the default value from define variable?

Might also be worth adding the test I mentioned in a comment to ensure we have the right logic for whether to evaluate a defineVariable defaultValue, or just use the value it evaluated to when parse/unaprse started.